server/service/handler.go

```diff
@@ -710,7 +710,7 @@ func attachFleetAPIRoutes(r *mux.Router, svc fleet.Service, config config.FleetC
 	mdmAppleMW.POST("/api/_version_/fleet/mdm/apple/profiles/preassign", preassignMDMAppleProfileEndpoint, preassignMDMAppleProfileRequest{})
 	mdmAppleMW.POST("/api/_version_/fleet/mdm/apple/profiles/match", matchMDMApplePreassignmentEndpoint, matchMDMApplePreassignmentRequest{})
 
-	mdmAnyMW := ue.WithCustomMiddleware(mdmConfiguredMiddleware.VerifyAppleOrWindowsMDM())
+	mdmAnyMW := ue.WithCustomMiddleware(mdmConfiguredMiddleware.VerifyAnyMDM())
```

Comment:
I think it could be nice to add a super small integration tests, that hit one of these (`mdmAnyMW`) endpoints, without any MDM enabled, to see the `ErrMDMNotConfigured` error, but then enabling every MDM one by one, to ensure we don't see that specific error (shouldn't matter if the later logic fails).

I know it's not a complete solution, but having a single endpoint using the MW tested is better than nothing.